### ISSUE TITLE
Linking github issues?

### ISSUE THREAD

**Issue by tony-johnson** [DarkEnergyScienceCollaboration/Twinkles#185]
This is a random question rather than an "issue", but I am not sure where else to ask this question: Is there any way to put links between github issues? For example for Twinkles I would like to create a "Reprocess all phosim runs" issue, and then link each issue which needs to be fixed before this can be done. This is something I do all of the time with JIRA and so far really miss with GitHub issues.

**Comment by cwwalter** [comment]
Just start to type a "#" and then number or text to bring it up.  Then select and it will put a link in for you.  Or maybe you are asking something more complicated, like actually putting in dependencies?

**Comment by brianv0** [comment]
You can also cross-repo, but it doesn't have autocomplete: DarkEnergyScienceCollaboration/ComputingInfrastructure#13

**Comment by tony-johnson** [comment]
Right I know I can put links in comments, but I want more than that. In JIRA I can actually create link types, like "Is Required By", "Is Related To", "Must be fixed before" etc. Then for any issue I can see what needs to be done before that issue can be completed (and then those issues may in turn have their own requirements). Links within comments does not seem an adequate substitute for this functionality.

**Comment by cwwalter** [comment]
Right... this kind of dependency doesn't exist.  

@connolly got us access to a JIRA type story board overlay (https://www.zenhub.io).  I'm not sure if it also gives that kind of extension.

**Comment by drphilmarshall** [comment]
As far as I know, ZenHub does not enable linking of issues in this way. The closest thing that could be useful for us is their ["Burndown Charts"](https://www.zenhub.io/burndownChart) - but its still all just labelled issues.

One thing we could try is making some issues be "epics":  "Reprocess all phosim runs" would be a good example. In an epic, the first issue comment is special: if you make a list of check boxes in this comment box, a little progress bar for that issue is rendered in the main issue list. All issue comments are editable, so the list can be grown with time if necessary. Then, each checklist item can be issued, and the new issue linked back to the epic (again by editing the check list in the first comment box). For clarity, we could make an epics label, so we can see them all easily. Shall we give this a try?

**Comment by tony-johnson** [comment]
Trying this sounds fine. To clarify, I do not need to do anything special to create an "epic", except create an issue, optionally label it as epic, and add a set of check boxes in the first comment (like this: https://github.com/DarkEnergyScienceCollaboration/ComputingInfrastructure/issues/14 I assume).

**Comment by drphilmarshall** [comment]
That's right - although notice that in that example Andy and I are not
using the issue as an epic, really, because we are not editing the first
comment. In the main list at
https://github.com/DarkEnergyScienceCollaboration/ComputingInfrastructure/issues
you can see that issue 14 is listed as having "0 of 3" tasks done. Here's a
real epic, that I just made for you:

https://github.com/DarkEnergyScienceCollaboration/ComputingModel/issues/99

**Comment by connolly** [comment]
You can also take a look at  a look at https://github.com/DarkEnergyScienceCollaboration/Twinkles/issues/41 (I just reopened it for you) which was used as an epic (you might be able to see it in zenhub - I havent checked)

**Comment by drphilmarshall** [comment]
Yes, that was a good one. Although I see that I closed it before the
checklist was complete! Tut tut.

**Comment by drphilmarshall** [comment]
Nice epic at #187, @tony-johnson ! I labelled it attractively for you :-) Feel free to close this if you like.
